stan/math/prim/arr/fun/promote_elements.hpp

```diff
@@ -3,6 +3,7 @@
 
 #include <stan/math/prim/scal/fun/promote_elements.hpp>
 #include <vector>
+#include <cstddef>
```

Comment:
Oddly, it was not passing on my local compiler either with g or clang (`clang version 3.8.1-12ubuntu1 (tags/RELEASE_381/final)` and `g++ (Ubuntu 6.2.0-5ubuntu12) 6.2.0 20161005` with lastest "stable" Ubuntu) complaining about `size_t` not being defined, but I can change it back if you wish, it might just be an Ubuntu bug anyway.